spacy_llm/models/hf/base.py

```diff
@@ -4,41 +4,40 @@
 
 from thinc.compat import has_torch_cuda_gpu
 
-from ....compat import has_accelerate, has_torch, has_transformers, torch
+from ...compat import has_accelerate, has_torch, has_transformers, torch
 
 # Type of prompts returned from Task.generate_prompts().
 _PromptType = TypeVar("_PromptType")
 # Type of responses returned from query function.
 _ResponseType = TypeVar("_ResponseType")
 
 
-class HuggingFaceBackend(abc.ABC):
-    """Backend for HuggingFace models."""
+class HuggingFace(abc.ABC):
+    """Base class for HuggingFace model classes."""
 
     def __init__(
         self,
-        model: str,
+        name: str,
         config_init: Optional[Dict[str, Any]],
         config_run: Optional[Dict[str, Any]],
     ):
-        """Initializes Backend instance.
+        """Initializes HF model instance.
         query (Callable[[Any, Iterable[_PromptType]], Iterable[_ResponseType]]): Callable executing LLM prompts when
             supplied with the `integration` object.
-        model (str): Name of HF model to load.
+        name (str): Name of HF model to load (without account name).
         config_init (Optional[Dict[str, Any]]): HF config for initializing the model.
         config_run (Optional[Dict[str, Any]]): HF config for running the model.
         inference_config (Dict[Any, Any]): HF config for model run.
         """
-        self._model_name = model
+        self._hf_model_id = f"{self.get_hf_account()}/{name}"
         self._config_init, self._config_run = self.compile_default_configs()
         if config_init:
             self._config_init = {**self._config_init, **config_init}
         if config_run:
             self._config_run = {**self._config_run, **config_run}
 
         # Init HF model.
-        HuggingFaceBackend.check_installation()
-        self.check_model()
```

Comment:
Why remove this `check_model` call & functionality?

Reply:
I considered it redundant as we only offer per-model instantiation now. On the other hand, if users instantiate a HF model in the code, it might still be helpful (because then the model name isn't hardcoded)...what do you think?

Reply:
I'm leaning towards pushing users to use the registered functions (they can still call e. g. `openai_gpt_4()` if they don't want to use the config system) and not offering support for instantiating `HuggingFaceModel` instances directly.

Reply:
There's still a list of names that are supported though, right? I'm thinking of use-cases where someone writes in the config
```
[components.llm.model]
@llm_models = "spacy.gpt-3-5.v1"
gpt-3.5-turbot
```

Reply:
I thought it was fine to have this as a Python utility, that directly queries the internals instead of having to manually look up in the docs? I can imagine the Python function to be useful to developers.

Reply:
Reintroduced in https://github.com/explosion/spacy-llm/pull/176/commits/70fa93286f97ad351371e9a986a209248664f331.